### ISSUE TITLE
"open : no such file or directory" during the creation instance with azure driver

### ISSUE THREAD

**Issue by dimorinny** [docker/machine#3465]
After re-installation OS X 10.11.5 I got a problem when creating docker-machine node with azure driver. 
When I am using this command:
```
docker-machine --debug create \
    --driver azure \
    --azure-subscription-id 3281883a-27db-4b61-b392-a91a0dcf4c6c \
    --azure-size Basic_A0 \
    --azure-location "West Europe" \
    test
```
I have this debug output:
```
Docker Machine Version:  0.7.0, build a650a40
Found binary path at /usr/local/bin/docker-machine
Launching plugin server for driver azure
Plugin server listening at address 127.0.0.1:64185
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(flag-lookup) Calling .GetMachineName
(flag-lookup) Calling .DriverName
(flag-lookup) Calling .GetCreateFlags
Found binary path at /usr/local/bin/docker-machine
Launching plugin server for driver azure
Plugin server listening at address 127.0.0.1:64189
() Calling .GetVersion
Using API Version  1
() Calling .SetConfigRaw
() Calling .GetMachineName
(test) Calling .GetMachineName
(test) Calling .DriverName
(test) Calling .GetCreateFlags
(test) Calling .SetConfigFromFlags
(test) DBG | Set configuration from flags.
Running pre-create checks...
(test) Calling .PreCreateCheck
(test) DBG | Looking up AAD Tenant ID.  subs="3281883a-27db-4b61-b392-a91a0dcf4c6c"
(test) DBG | Tenant ID loaded from file: /Users/Dimorinny/.docker/machine/credentials/azure/3281883a-27db-4b61-b392-a91a0dcf4c6c.tenantid
(test) DBG | Found AAD Tenant ID.  tenant="652d6fe7-45ba-491c-9780-a4ecfdb9de61" subs="3281883a-27db-4b61-b392-a91a0dcf4c6c"
(test) DBG | Loading auth token from file  path="/Users/Dimorinny/.docker/machine/credentials/azure/652d6fe7-45ba-491c-9780-a4ecfdb9de61.json"
(test) DBG | Auth token found in file.  path="/Users/Dimorinny/.docker/machine/credentials/azure/652d6fe7-45ba-491c-9780-a4ecfdb9de61.json"
(test) DBG | Validating the token.
(test) DBG | Azure token expired. Saving the refreshed token...
(test) DBG | Saved token to file.
(test) DBG | Azure request  method="GET" request="https://management.azure.com/subscriptions?api-version=2015-11-01"
(test) DBG | Azure response  status="200 OK" method="GET" request="https://management.azure.com/subscriptions?api-version=2015-11-01" x-ms-request-id="278dd9a7-500f-41f6-9886-3d5b96cce4d2"
(test) DBG | Azure response  status="200 OK" method="GET" request="https://management.azure.com/subscriptions?api-version=2015-11-01" x-ms-request-id="278dd9a7-500f-41f6-9886-3d5b96cce4d2"
(test) DBG | Token works.
Error with pre-create check: "azure.ServicePrincipalToken#WithAuthorization: Failed to refresh Service Principal Token for request to https://management.azure.com/subscriptions/3281883a-27db-4b61-b392-a91a0dcf4c6c/providers?api-version=2015-11-01: StatusCode=0 -- Original Error: Manually created ServicePrincipalToken does not contain secret material to retrieve a new access token."
open : no such file or directory
notifying bugsnag: [Error with pre-create check: "azure.ServicePrincipalToken#WithAuthorization: Failed to refresh Service Principal Token for request to https://management.azure.com/subscriptions/3281883a-27db-4b61-b392-a91a0dcf4c6c/providers?api-version=2015-11-01: StatusCode=0 -- Original Error: Manually created ServicePrincipalToken does not contain secret material to retrieve a new access token."]
```

**Comment by dimorinny** [comment]
I compiled docker-machine from HEAD manually and this problem was "solved". But now, when I trying to create new instance I had problem with certs. Here's end part of my docker-machine's create debug log. I am confused by line: "open : no such file or directory".

```
(test13) DBG | Matching pattern for code="ProvisioningState/succeeded"
(test13) DBG | Matching pattern for code="PowerState/running"
(test13) DBG | Power state found.
(test13) DBG | Determined Azure PowerState="running", docker-machine state="Running"
(test13) DBG | Machine IP address resolved to: 40.64.209.171
(test13) DBG | Machine URL is resolved to: tcp://40.64.209.171:2376
Reading CA certificate from /Users/Dimorinny/.docker/machine/certs/ca.pem
Reading server certificate from /Users/Dimorinny/.docker/machine/machines/test13/server.pem
Reading server key from /Users/Dimorinny/.docker/machine/machines/test13/server-key.pem
Error creating machine: Error checking the host: Error checking and/or regenerating the certs: There was an error validating certificates for host "40.68.209.171:2376": remote error: bad certificate
You can attempt to regenerate them using 'docker-machine regenerate-certs [name]'.
Be advised that this will trigger a Docker daemon restart which will stop running containers.

open : no such file or directory
notifying bugsnag: [Error creating machine: Error checking the host: Error checking and/or regenerating the certs: There was an error validating certificates for host "40.68.209.171:2376": remote error: bad certificate
You can attempt to regenerate them using 'docker-machine regenerate-certs [name]'.
Be advised that this will trigger a Docker daemon restart which will stop running containers.
```

**Comment by dimorinny** [comment]
I found what was the problem. That problem was in incorrect system time :)
